src/cockpit/packages.py

```diff
@@ -192,11 +193,18 @@ def relative_to(self, path):
         assert x.startswith(y)
         return Path(x[len(y):])
 
+# TODO: This doesn't yet implement the slighty complicated checksum
+# scheme of the C bridge (see cockpitpackages.c) to support caching
+# and reloading.
+
+class Packages(bus.Object, interface='cockpit.Packages'):
+    manifests = bus.Interface.Property('s', value="{}")
 
-class Packages:
     def __init__(self):
+        super().__init__()
         self.packages = {}
         self.load_packages()
+        self.ignore_next_reload_hint = True
```

Comment:
Why?

Reply:
When the shell loads, it always calls "ReloadHint".  The idea is that we want to reload packages when the user reloads the browser, so that any changes done on the filesystem (maybe by installing more cockpit-foo packages) will show up in the menu.

However, we don't want to reload packages when Cockpit is opened for the first time.

I think the Shell could get smarter and call "Reload", but only when it is actually being reloaded by the user.  I think there are ways to detect that. I didn't know anything about that when implementing package reloading way back when...

And yeah, this needs documenting....

Reply:
Documented in the code.